src/options.rs

```diff
@@ -118,6 +120,23 @@ impl SftpOptions {
         self.read_end_buffer_size
             .unwrap_or_else(|| NonZeroUsize::new(1024).unwrap())
     }
+
+    /// Set the write buffer limit for tokio compat file.
+    /// If the write buffer size is larger than the limit in tokio compat file,
+    /// then the buffer will be flushed and following async write operation will
+    /// be pending until flush operation is completed.
```

Comment:
I think it would be better to only flush some of the buffer until it has enough space for the current buffer, instead of flushiing the all buffers at once, which would take quite some time.

Reply:
It seems difficult to get flushed buffer size, because `poll_flush` interact with `AwaitableStatusFuture` directly. I have no idea for this.

Reply:
[`AwaitableStatusFuture`](https://docs.rs/openssh-sftp-client-lowlevel/latest/openssh_sftp_client_lowlevel/struct.AwaitableStatusFuture.html) is simply a `Future` with Output `Result<(Id<Buffer>, ()), Error>`, not very difficult to use.

You can simply trigger flushing in `poll_write` when full:

```
        // Flush only if there is pending awaitable writes
        if this.inner.need_flush {
            // Only flush if there are pending requests
            if this.inner.auxiliary().get_pending_requests() != 0 {
                this.inner.auxiliary().trigger_flushing();
            }
            this.inner.need_flush = false;
        }

        if this.cancellation_future.poll(cx).is_ready() {
            return Poll::Ready(Err(sftp_to_io_error(cancel_error())));
        }

        // And then wait on the first future only
        ...
```

You would also need to embed `usize` of the write into `write_futures`.

Reply:
Would also like u to update the doc here